autospec/buildreq.py

```diff
@@ -281,6 +281,44 @@ def _get_desc_field(field, desc):
     return val
 
 
+# FIXME: this list should be autodetected
+def _is_r_provide(pkg):
+    """List of packages provided by R, serving as blacklist for CRAN deps."""
+    provides = [
+        "KernSmooth",
+        "MASS",
+        "Matrix",
+        "base",
+        "boot",
+        "class",
+        "cluster",
+        "codetools",
+        "compiler",
+        "datasets",
+        "foreign",
+        "grDevices",
+        "graphics",
+        "grid",
+        "lattice",
+        "methods",
+        "mgcv",
+        "nlme",
+        "nnet",
+        "parallel",
+        "rpart",
+        "spatial",
+        "splines",
+        "stats",
+        "stats4",
+        "survival",
+        "tcltk",
+        "tools",
+        "translations",
+        "utils",
+    ]
+    return pkg in provides
+
+
```

Comment:
Since this is in a loop I wonder if a _get_r_provides function that returns a **set** (and is only called once) that is then used in the loop would be a better way of doing this.

Reply:
Yes, good idea... v2 patch will appear shortly.